### PR TITLE
Add imprint and privacy policy support

### PR DESCRIPTION
**Pull request by VicDeo**
OC version-agnostic :)
Ref #1

### CODE REVIEW

File: defaults.php [hunk 1]
```diff
@@ -116,14 +116,36 @@ public function getLogoClaim() {
...
+	}
+	
 	/**
 	 * Returns short version of the footer
 	 * @return string short footer
 	 */
 	public function getShortFooter() {
 		$footer = '© 2017 <a href="'.$this->getBaseUrl().'" target="_blank\">'.$this->getEntity().'</a>'.
 			'<br/>' . $this->getSlogan();
+			if ($this->getImprintUrl() !== '') {
```
Comment (PVince81) [inline defaults.php]: an update of the 2017 date is in order ? :smile:

**Review by PVince81 — DISMISSED** [review]
:+1:

File: defaults.php [hunk 2]
```diff
@@ -134,7 +156,13 @@ public function getShortFooter() {
 	public function getLongFooter() {
 		$footer = '© 2017 <a href="'.$this->getBaseUrl().'" target="_blank\">'.$this->getEntity().'</a>'.
 			'<br/>' . $this->getSlogan();
+			if ($this->getImprintUrl() !== '') {
+				$footer .= '<span class="nowrap"> | <a href="' . $this->getImprintUrl() . '" target="_blank">' . $this->l->t('Imprint') . '</a></span>';
```
Comment (VicDeo) [inline defaults.php]: `$this->l` is undefined

File: defaults.php [hunk 3]
```diff
@@ -134,7 +156,13 @@ public function getShortFooter() {
...
+			}
 
+			if ($this->getPrivacyPolicyUrl() !== '') {
+				$footer .= '<span class="nowrap"> | <a href="'. $this->getPrivacyPolicyUrl() .'" target="_blank">'. $this->l->t('Privacy Policy') .'</a></span>';
```
Comment (VicDeo) [inline defaults.php]: the same as above :/

File: defaults.php [hunk 4]
```diff
@@ -116,14 +116,36 @@ public function getLogoClaim() {
...
+	
 	/**
 	 * Returns short version of the footer
 	 * @return string short footer
 	 */
 	public function getShortFooter() {
 		$footer = '© 2017 <a href="'.$this->getBaseUrl().'" target="_blank\">'.$this->getEntity().'</a>'.
 			'<br/>' . $this->getSlogan();
+			if ($this->getImprintUrl() !== '') {
+				$footer .= '<span class="nowrap"> | <a href="' . $this->getImprintUrl() . '" target="_blank">' . $this->l->t('Imprint') . '</a></span>';
```
Comment (VicDeo) [inline defaults.php]: `$this->l` is undefined

File: defaults.php [hunk 5]
```diff
@@ -116,14 +116,36 @@ public function getLogoClaim() {
...
+			}
 
+			if ($this->getPrivacyPolicyUrl() !== '') {
+				$footer .= '<span class="nowrap"> | <a href="'. $this->getPrivacyPolicyUrl() .'" target="_blank">'. $this->l->t('Privacy Policy') .'</a></span>';
```
Comment (VicDeo) [inline defaults.php]: `$this->l` is undefined

**Comment by PVince81** [comment]
hmm, so need a way to inject the l10n object ?

**Comment by VicDeo** [comment]
@PVince81 I'm not sure it's a good idea to increase complexity level with DI in OC_Theme 
maybe just add 
```
protected function getL10n() {
}
```

and replace `$this->l` with `$this->getL10n`

**Comment by PVince81** [comment]
That's sad. This means even more copy-paste code ? If there is no other way then ok, will need to be in the docs.

**Comment by VicDeo** [comment]
@PVince81 updated

**Comment by VicDeo** [comment]
@pmaier1 done from my POV. Unless somebody will ask for changes.

**Review by PVince81 — APPROVED** [review]
As discussed :+1: 

There is currently no better way.

**Pull request merged**